llm_perf/common/benchmark_runner.py

```diff
@@ -0,0 +1,115 @@
+            )
+        else:
+            raise ValueError("Either both MACHINE and SUBSET should be set for benchmarking or neither for debugging")
+
+        self.attention_configs = self._get_attention_configs()
+        self.weights_configs = self._get_weights_configs(self.subset)
+
+        self.logger.info(f"len(OPEN_LLM_LIST): {len(OPEN_LLM_LIST)}")
+        self.logger.info(f"len(PRETRAINED_OPEN_LLM_LIST): {len(PRETRAINED_OPEN_LLM_LIST)}")
+        self.logger.info(f"len(CANONICAL_PRETRAINED_OPEN_LLM_LIST): {len(CANONICAL_PRETRAINED_OPEN_LLM_LIST)}")
+
+    @abstractmethod
+    def _get_weights_configs(self, subset: str) -> Dict[str, Dict[str, Any]]:
+        raise NotImplementedError("This method should be implemented in the child class")
+
+    @abstractmethod
+    def _get_attention_configs(self) -> List[str]:
+        raise NotImplementedError("This method should be implemented in the child class")
+
+    def is_benchmark_supported(self, weights_config: str, attn_implementation: str) -> bool:
+        return True
+
+    def run_benchmarks(self):
+        os.environ["LOG_TO_FILE"] = "0"
+        os.environ["LOG_LEVEL"] = "INFO"
+        setup_logging(level="INFO", prefix="MAIN-PROCESS")
+
+        models_attentions_weights = list(
+            product(CANONICAL_PRETRAINED_OPEN_LLM_LIST, self.attention_configs, self.weights_configs.keys())
+        )
+
+        self.logger.info(
+            f"Running a total of {len(models_attentions_weights)} benchmarks, "
+            f"with {len(CANONICAL_PRETRAINED_OPEN_LLM_LIST)} models, "
+            f"{len(self.attention_configs)} attentions implementations "
+            f"and {len(self.weights_configs)} weights configurations."
+        )
+
+        for model, attn_implementation, weights_config in models_attentions_weights:
+            self.run_benchmark(model, attn_implementation, weights_config)
+
+    def is_benchmark_conducted(self, push_repo_id, subfolder):
+        try:
+            report = BenchmarkReport.from_pretrained(repo_id=push_repo_id, subfolder=subfolder)
+            if "traceback" in report.to_dict():
+                return False
+            else:
+                return True
+        except Exception:
+            return False
+
+    def run_benchmark(self, model: str, attn_implementation: str, weights_config: str):
+        benchmark_name = f"{weights_config}-{attn_implementation}"
+        subfolder = f"{benchmark_name}/{model.replace('/', '--')}"
+
+        if not self.is_benchmark_supported(weights_config, attn_implementation):
+            self.logger.info(f"Skipping benchmark {benchmark_name} with model {model} since it is not supported")
+            return
+
+        if self.is_benchmark_conducted(self.push_repo_id, subfolder):
+            self.logger.info(f"Skipping benchmark {benchmark_name} with model {model} since it was already conducted")
+            return
```

Comment:
won't hold for other backends as their configuration will be in term of optimizations and quantization schemes to use